src/System.Management.Automation/security/Authenticode.cs

```diff
@@ -363,10 +363,23 @@ private static Signature GetSignatureFromCatalog(string filename)
                             DWORD error = GetErrorFromSignatureState(sigInfo.nSignatureState);
 
                             X509Certificate2 cert = null;
+
                             if (ppCertContext != IntPtr.Zero)
                             {
                                 cert = new X509Certificate2(ppCertContext);
-                                signature = new Signature(filename, error, cert);
+
+                                // Get the time stamper certificate if available
+                                IntPtr pProvSigner = IntPtr.Zero;
+                                X509Certificate2 timestamperCert = null;
+                                TryGetProviderSigner(phStateData, ref pProvSigner, ref timestamperCert);
+                                if (timestamperCert != null)
+                                {
+                                    signature = new Signature(filename, error, cert, timestamperCert);
+                                }
+                                else
+                                {
+                                    signature = new Signature(filename, error, cert);
+                                }
```

Comment:
Once the parameters `pProvSigner` and `timestamperCert` are changed to `out` parameters, you can use the new language feature in C# for out parameters:
```CSharp
if (int.TryParse(input, out int result))
    WriteLine(result);
else
    WriteLine("Could not parse input");
```
So this code block can be rewritten to:
```CSharp
if (TryGetProviderSigner(phStateData, out IntPtr pProvSigner, out X509Certificate2 timestamperCert) && timestamperCert != null)
{
    signature = new Signature(filename, error, cert, timestamperCert);
}
else
{
    signature = new Signature(filename, error, cert);
}
```